Modules/Devices/test/Devices.Application.Tests/Tests/Tiers/Commands/DeleteTier/HandlerTests.cs

```diff
@@ -0,0 +1,90 @@
+using Enmeshed.BuildingBlocks.Application.Abstractions.Infrastructure.EventBus;
+using Enmeshed.DevelopmentKit.Identity.ValueObjects;
+using FakeItEasy;
+using FluentAssertions;
+using Microsoft.Extensions.Logging;
+using Xunit;
+using ApplicationException = Enmeshed.BuildingBlocks.Application.Abstractions.Exceptions.ApplicationException;
+
+namespace Backbone.Modules.Devices.Application.Tests.Tests.Tiers.Commands.DeleteTier;
+public class HandlerTests
+{
+    private readonly ITiersRepository _tierRepository;
+    private readonly IEventBus _eventBus;
+    private readonly Handler _handler;
+
+    public HandlerTests()
+    {
+        _tierRepository = A.Fake<ITiersRepository>();
+        _eventBus = A.Fake<IEventBus>();
+        _handler = CreateHandler();
+    }
+
+    private Handler CreateHandler()
+    {
+        var logger = A.Fake<ILogger<Handler>>();
+        return new Handler(_tierRepository, logger, _eventBus);
+    }
+
+    [Fact]
+    public async Task Cannot_Delete_Basic_Tier()
+    {
+        // Arrange
+        var basicTierName = TierName.Create("basic-tier").Value;
+        var basicTier = new Tier(basicTierName);
+        A.CallTo(() => _tierRepository.GetBasicTierAsync(A<CancellationToken>._)).Returns(Task.FromResult(basicTier));
+        A.CallTo(() => _tierRepository.FindById(basicTier.Id, A<CancellationToken>._)).Returns(Task.FromResult(basicTier));
+
+        // Act
+        var acting = async () => await _handler.Handle(new DeleteTierCommand(basicTier.Id), CancellationToken.None);
+
+        // Assert
+        await acting.Should().ThrowAsync<ApplicationException>().WithErrorCode("error.platform.validation.device.basicTierCannotBeDeleted");
+    }
+
+    [Fact]
+    public async Task Cannot_Delete_Tier_With_Associated_Identities()
+    {
+        // Arrange
+        var basicTier = new Tier(TierName.Create("basic-tier").Value);
+        var someIdentity = new Identity(
+            TestDataGenerator.CreateRandomDeviceId(),
+            TestDataGenerator.CreateRandomIdentityAddress(),
+            TestDataGenerator.CreateRandomBytes(),
+            TestDataGenerator.CreateRandomTierId(),
+            1);
+        var anotherTier = new Tier(TierName.Create("random-tier").Value) { Identities = new List<Identity>() { someIdentity } };
+
+        A.CallTo(() => _tierRepository.GetBasicTierAsync(A<CancellationToken>._)).Returns(Task.FromResult(basicTier));
+        A.CallTo(() => _tierRepository.FindById(anotherTier.Id, A<CancellationToken>._)).Returns(Task.FromResult(anotherTier));
+
+        // Act
+        var acting = async () => await _handler.Handle(new DeleteTierCommand(anotherTier.Id), CancellationToken.None);
+
+        // Assert
+        await acting.Should().ThrowAsync<ApplicationException>().WithErrorCode("error.platform.validation.device.usedTierCannotBeDeleted");
+    }
+
+    [Fact]
+    public async Task Delete_Non_Basic_Tier_Without_Identities_Publishes_IntegrationEvent()
```

Comment:
What if there were 10 more preconditions for a successful deletion? Would you add all of them to this test name?